Geometry/src/LUXEGeometry.cpp

```diff
@@ -1,125 +1,112 @@
-    //
-    G4double worldSizeXY = 50;
-    G4double worldSizeZ = 50;
-    G4Material* worldMat = nist->FindOrBuildMaterial("G4_Galactic");
-    
-    auto solidWorld = new G4Box("World", 0.5 * worldSizeXY, 0.5 * worldSizeXY,
-                                0.5 * worldSizeZ);
-    
-    auto logicWorld = new G4LogicalVolume(solidWorld, worldMat, "World");
-    
-    auto physWorld = new G4PVPlacement(nullptr, G4ThreeVector(), logicWorld,
-                                        "World", nullptr, false, 0, checkOverlaps);
-    
-    G4Material* material = nist->FindOrBuildMaterial("G4_He");
-    
-    std::vector<std::string> names;
-    for (int nArm = 0; nArm < nArms; nArm++) {
-        for (int nLayer = 0; nLayer < nLayers; nLayer++) {
-            for (int nChip = 0; nChip < nChips; nChip++) {
-                int sign = (nArm == 0) ? 1 : -1;
-                double posX = sign * (armOffset + nChip);
-                double posY = 0;
-                double posZ = nLayer;
-                G4ThreeVector pos = G4ThreeVector(posX, posY, posZ);
-                G4ThreeVector dims = G4ThreeVector(cellDimX, cellDimY, cellDimZ);
-        
-                int cellId = nChips * nLayers * nArm + nChips * nLayer + nChip;
-                std::string name = "cell" + std::to_string(cellId);
-        
-                names.push_back(name);
-        
-                // Box cell
-                auto solidCell = new G4Box(name, dims[0], dims[1], dims[2]);
-        
-                G4LogicalVolume* cellLogical =
-                    new G4LogicalVolume(solidCell, material, "cellSensitive");
-        
-                new G4PVPlacement(nullptr, pos, cellLogical, name, logicWorld, false, 0,
-                                true);
-            }
-        }
-    }
-
-    // Write GDML file
-    G4GDMLParser parser;
-    parser.SetOutputFileOverwrite(true);
-    parser.Write(gdmlPath, physWorld);
-
-    return std::make_tuple(physWorld, names, gdmlPath);
-}
+std::shared_ptr<Acts::Experimental::LayerStructureBuilder> makeLayerBuilder(std::string gdmlPath,
+                                                                            std::vector<std::string> names,
+                                                                            Acts::GeometryContext gctx,
+                                                                            Acts::ActsScalar zMin,
+                                                                            Acts::ActsScalar zMax) {
+    auto spCfg = Acts::Experimental::Geant4SurfaceProvider<3>::Config();
+    spCfg.gdmlPath = gdmlPath;
+    spCfg.surfacePreselector =
+            std::make_shared<Acts::Geant4PhysicalVolumeSelectors::NameSelector>(names,
+                                                                                true);
+
+    auto kdtDOpt = Acts::Experimental::Geant4SurfaceProvider<3>::kdtOptions();
+    kdtDOpt.range = Acts::RangeXD<3, Acts::ActsScalar>();
+    kdtDOpt.range[0].set(zMin, zMax);
```

Comment:
Also, the range is initialized to be degenerate by default, i.e. `xMin` > `xMax.` If the range is degenerate, `KDTree` does not run the selection. Are you sure that it works?